app/grandchallenge/algorithms/templates/algorithms/job_list_row.html

```diff
@@ -27,13 +27,7 @@
 
 <split></split>
 
-<span class="badge badge-{{ object.status_context }}">
-    {% if object.animate %}
-        <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>
-    {% endif %}
-    {{ object.get_status_display }}{% if object.status == object.SUCCESS and object.stderr %},
-        with warnings{% endif %}
-</span>
+<div hx-get="{% url 'algorithms:job-status-badge-detail' slug=object.algorithm_image.algorithm.slug pk=object.pk %}" hx-trigger="revealed" hx-swap="outerHTML"></div>
```

Comment:
Is there a way to do this without htmx on load? Other wise we will hammer the server with these requests.

Reply:
I'm not sure yet. I did it this way to get the random delay in the context generated by the view. One way I could do it is by generating the random delay in the template itself, which is also possible (although not very pretty). 
That would be the only way to avoid multiple requests I think...

Reply:
Why can't you use `{% include "algorithms/job_status_badge_detail.html" %}` here? That way you would avoid multiple requests for completed jobs.

Then, for generating a random delay rather than setting the context on the view you could add a [custom template tag](https://docs.djangoproject.com/en/5.1/howto/custom-template-tags/#writing-custom-template-tags) for `randint` so that you could do:

```
hx-trigger="every {% randint 20 40 %}s"
```

in your template.

Reply:
Updated the example to use delay as the outerHTML is being replaced here, see https://htmx.org/attributes/hx-trigger/. Would that still work with the new badges brought in?

Reply:
Also, interestingly in the HTMX docs there is a queue option, so we might not need to jitter the requests at all, could you test with a straight up 30s delay with multiple buttons on a page and check the timing diagram?

Reply:
Thanks for the pointer to custom template tags. That is much nicer than what I had. 

I replaced `every` with a `delay` modifier on the load event. `hx-trigger="load delay:{% randint 20 40 %}"`
(This does not work with the "revealed" event btw. If I use that, delay is ignored and the request triggers constantly.)

Reply:
> Also, interestingly in the HTMX docs there is a queue option, so we might not need to jitter the requests at all, could you test with a straight up 30s delay with multiple buttons on a page and check the timing diagram?

I read about the queue modifier in hx-trigger and hx-sync. Is that the option you mean? 

With ten badges with a delay of 30s I see the following:

![image](https://github.com/user-attachments/assets/4ae5b4ef-2084-4b66-90c0-b77d260c724a)

![image](https://github.com/user-attachments/assets/3a854f10-ce9d-40bd-846e-d75609eb28cc)

Reply:
Ah perfect. HTMX is great and we do not need to jitter then, super!